src/System.Private.ServiceModel/tests/Common/Infrastructure/xunit/SkippedTestCase.cs

```diff
@@ -0,0 +1,53 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+using Xunit.Sdk;
+
+namespace Infrastructure.Common
+{
+    /// <summary>Wraps another test case that should be skipped.</summary>
+    internal sealed class SkippedTestCase : IXunitTestCase
+    {
+        private readonly IXunitTestCase _testCase;
+        private readonly string _skippedReason;
+
+        internal SkippedTestCase(IXunitTestCase testCase, string skippedReason)
+        {
+            _testCase = testCase;
+            _skippedReason = skippedReason;
+        }
+
+        public string DisplayName { get { return _testCase.DisplayName; } }
+
+        public IMethodInfo Method { get { return _testCase.Method; } }
+
+        public string SkipReason { get { return _skippedReason; } }
+
+        public ISourceInformation SourceInformation { get { return _testCase.SourceInformation; } set { _testCase.SourceInformation = value; } }
+
+        public ITestMethod TestMethod { get { return _testCase.TestMethod; } }
+
+        public object[] TestMethodArguments { get { return _testCase.TestMethodArguments; } }
+
+        public Dictionary<string, List<string>> Traits { get { return _testCase.Traits; } }
+
+        public string UniqueID { get { return _testCase.UniqueID; } }
+
+        public void Deserialize(IXunitSerializationInfo info) { _testCase.Deserialize(info); }
+
+        public Task<RunSummary> RunAsync(
```

Comment:
I'd prefer not to deviate from using "Fact" or "Theory" in the name, because those are fundamental xunit concepts.  I'm open to renaming the prefix part (example: TestFact and TestTheory or WcfTestFact and WcfTheoryFact).  Thoughts?